spectralevents.py

```diff
@@ -195,16 +195,17 @@ def _energyvec(f, s, Fs, width=7.):
     sf = f / width
     st = 1 / (2 * np.pi * sf)
 
-    t = np.arange(-3.5 * st, 3.5 * st, dt)
-    m = _morlet(f, t, width)
+    t_single_side = np.arange(0., 3.5 * st, dt)
+    # mirror about 0; always an odd # of elements
+    times = np.r_[-t_single_side[-1:0:-1], t_single_side]
+    wavelet = _morlet(f, times, width)
 
-    y = np.convolve(s, m)
-    y = 2 * (dt * np.abs(y)) ** 2
-    lowerLimit = int(np.ceil(len(m) / 2))
-    upperLimit = int(len(y) - np.floor(len(m) / 2) + 1)
```

Comment:
To be consistent with the original matlab version, `lowerLimit` would have needed to be `int(np.ceil(len(m) / 2) - 1)` given Pythonic indexing. Similar for `upperLimit`. The matlab version wasn't totally correct though either....

Reply:
Yup - that would do it! This procedure for rounding down half the (odd) length looks good.